spec/models/debugs_bunny/debug_trace_spec.rb

```diff
@@ -32,6 +32,30 @@
     expect(debug_trace.created_at).to be_present
   end
 
+  # rubocop:disable RSpec/ExampleLength
+  it 'persists if the enclosing transaction is rolled back' do
+    error_klass = Class.new(StandardError)
+
+    guid = described_class.generate_guid
+    dump = 'Hello world!'
+
+    expect do
+      suppress(error_klass) do
+        described_class.transaction do
```

Comment:
why do you need two transactions?

Reply:
I don't necessarily need two transactions, but it's illustrative of what actually happens when the rollbacks occur. When stepping through the debugger:

1. The trace is created
1. The error is raised
1. The inner transaction is rolled back along with the trace 
1. The trace is recreated in `rollback_callback`
1. The error bubbles up to the outer transaction
1. The outer transaction is rolled back along with the recreated trace
1. The trace is recreated in `rollback_callback`

I wanted to ensure that this process results in 1 final trace (and not more or less than 1), regardless of how many nested transactions we are in.